crates/astria-proto/proto/astria/execution/v1alpha2/execution.proto

```diff
@@ -0,0 +1,107 @@
+syntax = 'proto3';
+
+package astria.execution.v1alpha2;
+
+import "google/protobuf/timestamp.proto";
+
+// The set of information which deterministic driver may to know about a given executed Block
+message Block {
+  // The block number
+  uint32 number = 1;
+  // The hash of the block
+  bytes hash = 2;
+  // Timestamp on the block, standardized to google protobuf standard.
+  google.protobuf.Timestamp timestamp = 3;
+}
+
+// These fields should be indexed on most block chains, and can be used to identify a block.
+message BlockIdentifier {
+  oneof identifier {
+    uint32 block_number = 1;
+    bytes block_hash = 2;
+  }
+}
+
+message GetBlockRequest {
+  BlockIdentifier identifier = 1;
+}
+message GetBlockResponse {
+    Block block = 1;
+}
+
+message BatchGetBlocksRequest {
+  repeated BlockIdentifier identifiers = 1;
+}
+message BatchGetBlocksResponse {
+  repeated Block blocks = 1;
+}
+
+
+// CreateBlockRequest contains all the information needed to create a new executed block.
+//
+// This information comes from previous execution blocks, as well as from sequencer blocks.
+message CreateBlockRequest {
```

Comment:
```suggestion
// ExecuteBlockRequest contains all the information needed to create a new executed block.
//
// This information comes from previous execution blocks, as well as from sequencer blocks.
message ExecuteBlockRequest {
```

I slightly prefer this, thoughts?

Reply:
All verbs chosen to be resource based: https://google.aip.dev/133. My thought here is that fundamentally we have two resources (Blocks, and a CommitmentState). Creating a block is what the API call does, executing is just a part of creating the new resource.